src/app/datasets/page.tsx

```diff
@@ -2,6 +2,67 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
-export default function DatasetPage() {
-  return <h1>Datasets</h1>;
+import ClientWrapper from "@/app/datasets/clientWrapper";
+import { datasetList, fieldDetailsGet } from "@/services/ckan/index.server";
+import { Field } from "@/services/ckan/types/fieldDetails.types";
+import { PackageSearchOptions } from "@/services/ckan/types/packageSearch.types";
+import { redirect } from "next/navigation";
+
+type DatasetPageProps = {
+  searchParams: Record<string, string>;
+};
+
+async function DatasetPage({ searchParams }: DatasetPageProps) {
+  if (!searchParams?.page) {
+    redirect("/datasets?page=1");
```

Comment:
question: I don't page load events in React, but this redirect may trigger twice the server side pagination/search. Can you please check?

Reply:
Good point, I haven't had much time but I did not find a clear answer. I will write it down to not forget it